CollateralAccount/CollateralAccount.md

```diff
@@ -0,0 +1,109 @@
+---
+CIP: ?
+Title: Collateral Key for HD Wallets
+Authors: Sebastien Guillemot <seba@dcspark.io>
+Comments-URI: https://forum.cardano.org/t/collateral-account-derivation/65879
+Status: Draft
+Type: Standards
+Created: 2021-06-29
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This document describes using a separate derivation path to ensure there is always sufficient collateral to execute UTXO smart contracts
+
+# Motivation
+
+Collateral input | As of Alonzo, transactions that call Plutus smart contracts are required to put up collateral to cover the potential cost of smart contract execution failure. Inputs used as collateral has the following properties:
+
+- Cannot contain any tokens (only ADA)
+- Cannot be a script address
+- Must be a UTXO input
+- Must be at least some percentage of the fee in the tx (concrete percentage decided by a protocol parameter)
+- Can be the same UTXO entry as used in non-collateral tx input
+- Is consumed entirely (no change) if the contract execution fails during phase 2 validation
+- Is not consumed is phase phase 2 validation succeeds
+
+Additionally, there cannot be more than *maxColInputs* (protocol parameter) inputs and the inputs have to cover a percentage of the fee defined by *collateralPercent* (protocol parameter)
+
+We therefore need a way for the wallet to handle picking which inputs to use for collateral
+
+## Naive solution
+
+The naive solution would do the following steps
+1. Add any non-collateral input that satisfies the constraints as collateral as well
+1. If collateral is still insufficient, look at the wallet UTXO for any other entry that can be added as collateral
+1. If there are not enough UTXO entries that can be used as collateral, check if we can rearrange the wallet UTXO to create enough UTXO entries that match the requirement
+1. If rearranging can unblock the wallet, prompt the user to send a transaction to rearrange their UTXO (and pay the tx fee associated with it) before sending their transaction. If rearranging can't unblock the wallet, tell the user they will need more ADA to call this smart contract
+
+This however, causes the following problems:
+1. This may cause users to risk more collateral than they are comfortable with. In general, software should run smart contracts locally to detect if a transaction would fail before sending and alert the user to avoid consuming the collateral, but relying on this is not ideal.
+1. Although the rearrange transaction can be created under the hood for in-software wallet, it will confused users when it shows up on their transaction history. For hardware wallets, it will require explicit approval from the user which is also possibly confusing.
```

Comment:
```suggestion
1. Although the rearrange transaction can be created under the hood for in-software wallet, it will confuse users when it shows up on their transaction history. For hardware wallets, it will require explicit approval from the user which is also possibly confusing.
```